src/task.rs

```diff
@@ -9,133 +9,158 @@ use esp_idf_sys::*;
 use crate::delay::TickType;
 use crate::interrupt;
 
-#[inline(always)]
-#[link_section = ".iram1.interrupt_task_do_yield"]
-pub fn do_yield() {
-    if interrupt::active() {
-        unsafe {
-            if let Some((yielder, arg)) = interrupt::get_isr_yielder() {
-                yielder(arg);
-            } else {
-                #[cfg(esp32c3)]
-                vPortYieldFromISR();
-
-                #[cfg(all(not(esp32c3), esp_idf_version_major = "4"))]
-                vPortEvaluateYieldFromISR(0);
-
-                #[cfg(all(not(esp32c3), not(esp_idf_version_major = "4")))]
-                _frxt_setup_switch();
-            }
-        }
-    } else {
-        unsafe {
-            vPortYield();
+/// OS task handle.
+pub struct Task {
```

Comment:
Can you compile with `--features edge-executor`? I'm relatively certain that this feature - as well as the `notify` module in `esp-idf-svc` is broken by these changes.

But in general I'm not really sure what this change is giving you really:
- Most of the methods in the `task` module do not take `&self`, as they operate on the "current" task. Why moving them then into a Task structure? 
- The one method that - after the change - can take `&self` - the `wait*notification` variant remains unsafe. So what gives?

The changes look to me as if they are more of a question of personal preference as to how stuff should be organized rather than addressing a real stopper problem. Or am I missing something?